mod/state-transition/pkg/core/state_processor_staking.go

```diff
@@ -83,14 +85,19 @@ func (sp *StateProcessor[
 	st BeaconStateT,
 	dep DepositT,
 ) error {
-	depositIndex, err := st.GetEth1DepositIndex()
-	if err != nil {
-		return err
+	var nextDepositIndex uint64
+	switch depositIndex, err := st.GetEth1DepositIndex(); {
+	case err == nil:
+		nextDepositIndex = depositIndex + 1
+	case sp.processingGenesis && err != nil:
+		// Eth1DepositIndex may have not been set yet
+		nextDepositIndex = 0
+	default:
+		// Failed retrieving Eth1DepositIndex outside genesis is an error
+		return fmt.Errorf("failed retrieving eth1 deposit index: %w", err)
```

Comment:
style nit: feels cleaner to move this logic into `st.GetEth1DepositIndex()`, which maybe takes in a `processingGenesis bool` flag?